packages/suite/src/views/wallet/transactions/TransactionList/useFetchTransactions.ts

```diff
@@ -0,0 +1,223 @@
+            setPagesFetched(page);
+        }
+    }, [pagesFetched, page]);
+
+    const isLastPage = pagesFetched >= pagesTotal;
+
+    useEffect(() => {
+        if (!fetchedAll && isLastPage) {
+            setFetchedAll(true);
+        }
+    }, [fetchedAll, isLastPage]);
+
+    const synchronize = useMemo(getSynchronize, [accountKey]);
+    const dispatch = useDispatch();
+
+    const fetchCommon = useCallback(
+        (
+            page: number,
+            options: {
+                recursive?: boolean;
+            } = {},
+        ) => {
+            if (options.recursive) {
+                // NOTE: when recurrsion is requested, load all the transactions along but don't wait for it
+                dispatch(fetchAllTransactionsForAccountThunk({ accountKey }));
+            }
+
+            return dispatch(
+                fetchTransactionsPageThunk({
+                    accountKey: account.key,
+                    page,
+                    perPage,
+                }),
+            );
+        },
+        [dispatch, account.key, perPage, accountKey],
+    );
+
+    const fetchPage = useCallback(
+        (
+            page: number,
+            options: {
+                recursive?: boolean;
+                noLoading?: boolean;
+            } = {},
+        ) => {
+            console.log('fetchPage:page', page, options);
+            synchronize(async () => {
+                setFetching(true);
+                await dispatch(
+                    fetchTransactionsPageThunk({
+                        accountKey: account.key,
+                        page,
+                        perPage,
+                        noLoading: Boolean(options.noLoading),
+                    }),
+                );
+                setFetching(false);
+            });
+        },
+        [account.key, dispatch, perPage, synchronize],
+    );
+
+    const fetchNext = useCallback(
+        () =>
+            synchronize(async () => {
+                if (fetchedAll) return;
+                setFetching(true);
+                await fetchCommon(pagesFetched + 1);
+                setFetching(false);
+                setPagesFetched(pagesFetched + 1);
+            }),
+        [synchronize, fetchCommon, pagesFetched, fetchedAll],
+    );
+
+    const fetchAll = useCallback(
+        () =>
+            synchronize(async () => {
+                if (fetchedAll) return;
+                setFetching(true);
+                await fetchCommon(pagesFetched + 1, {
+                    recursive: true,
+                });
+                setFetching(false);
+                setFetchedAll(true);
+            }),
+        [synchronize, fetchCommon, pagesFetched, fetchedAll],
+    );
+
+    return { fetchNext, pagesFetched, fetchPage, fetchAll, isFetching, fetchedAll };
+};
+
+export const useVisibleTransactionsRetriever = ({
+    account,
+    numberOfPagesRequested,
+}: {
+    account: Account;
+    numberOfPagesRequested: number;
```

Comment:
this number should be the number of the page in the UI.

When, for some reason, the FE filtering displays different number / paging of the transactions, we should treat the number in the UI as the one which should detemine how many "pages" of the transactions we should fetch.

As the "totals" and from that derive paging, doesn't correspond with the UI.